### ISSUE TITLE
PDB information for virtual function tables is defined incorrectly (Windows MSVC)

### ISSUE THREAD

**Issue by CySHell** [Vector35/binaryninja-api#3625]
**Version and Platform (required):**
- Binary Ninja Version: 3.2.3814-dev
- OS: Windows 
- OS Version: 11
- CPU Architecture: x64

**Bug Description:**
When defining types from a PDB file, a class' virtual table symbol is given a mock type that doesn't exist (see screenshot, type "SVR_BINDING_HANDLE::VTable" does not exist.
Since the type does not exist the data var is not recognized as a pointer to a function, and thus bv.get_data_refs_at(vtable_addr)
return 0 instead of the actual ref.
This is bad for algorithms that use this information for C++ RTTI parsing.

**Steps To Reproduce:**

1. open rpcrt4.dll with binja and load PDB info.
2. goto the "SVR_BINDING_HANDLE::`vftable'" data var in linear mode.
3. bv.get_data_refs(here)
4. If you set the type of this data var to a pointer (for instance "void*") then you will correctly get the data ref.

**Comment by CouleeApps** [comment]
Yep, can confirm. Looks like it sees the symbol for SVR_BINDING_HANDLE::`vftable' and thinks "oh there must be a vtable type for that," even though the pdb does not contain such a type. In these cases, we can either attempt to create a vtable type based on the data found in the binary, or just fall back. Considering the amount of guesswork involved to create a vtable type, I'm leaning more towards just falling back.

**Comment by CouleeApps** [comment]
Somewhere along the way, this was changed to use a void pointer. I thought the changes from #4574 would create an empty struct type for this, but that seems to have had no impact since the pointer to void has higher confidence than the structure (pointer!255 to void!0 vs struct!0, but the "pointer"-ness confidence wins because it's the base type... might need to be revisited later). I'm leaving this open to address this latter point, but the original issue you've run into with the xrefs should be resolved.
